pkg/fakerp/middleware.go

```diff
@@ -24,6 +24,10 @@ func (s *Server) validator(handler http.Handler) http.Handler {
 				http.Error(w, resp, http.StatusLocked)
 				return
 			}
+			defer func() {
```

Comment:
This change is going to break blocking DELETEs and PUTs asynchronously.

Reply:
can you please elaborate how this breaks things?

Reply:
When the request exits asynchronously, you release the lock too early while there is a background process that is actually running.

Reply:
So essentially we will block accepting PUT or DELETE requests even when those requests are being handled asynchronously? This definitey sounds like a spot for improvement. 

On the other hand, without releasing the lock, spurious DELETE or PUT requests will block other requests e.g. if I make a PUT request to `/` that will block all legitimate PUT requests for resources from the RP.

Will need to rework this.

Reply:
Yes, the current implementation does not account for managing different cluster deployments but that is probably a separate issue.

Reply:
Both CreateOrUpdate and Restore are synchronous and we don't block on deletes atm so this is ok for now.